### PR TITLE
fix Local externalTrafficPolicy is not respected for ipvs NodePort

### LINKED ISSUE #66062: Local externalTrafficPolicy is not respected for ipvs NodePort
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig network
/area kube-proxy
/area ipvs
/assign @honkiko

**What happened**:
When externalTrafficPolicy=Local, all endpoint addresses are added as RealServer for the NodePort VS.
When accessing NodePort, if the request is directed to an **remote**  pod, the response will not go back to the node because it's not masqueraded. Client will see a timeout.

**What you expected to happen**:
Only local endpoint addresses are added as RS for NodePort VS.

**How to reproduce it (as minimally and precisely as possible)**:
Start kube-proxy with --proxy-mode=ipvs and correct --hostname-override (same as node name).
Create an service with type=NodePort and externalTrafficPolicy=Local
Access the NodePort multiple times.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):  commit 0ff24a29c0 
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by honkiko**
**What this PR does / why we need it**:
Local externalTrafficPolicy is not respected for ipvs NodePort.
This PR fixes it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66062

**Special notes for your reviewer**:
Manually tested accessing NodePort with externalTrafficPolicy=Local  and externalTrafficPolicy=Cluster.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/proxy/ipvs/proxier.go [hunk 1]
```diff
@@ -1116,7 +1116,7 @@ func (proxier *Proxier) syncProxyRules() {
 				// There is no need to bind Node IP to dummy interface, so set parameter `bindAddr` to `false`.
 				if err := proxier.syncService(svcNameString, serv, false); err == nil {
 					activeIPVSServices[serv.String()] = true
-					if err := proxier.syncEndpoint(svcName, false, serv); err != nil {
+					if err := proxier.syncEndpoint(svcName, svcInfo.OnlyNodeLocalEndpoints, serv); err != nil {
```
Comment (m1093782566) [inline pkg/proxy/ipvs/proxier.go]: The 2nd parameter of `proxier.syncEndpoint()` is for indicating if we bind service address to dummy interface. I am not sure the change can fix your problems.

**Comment by honkiko** [comment]
in syncEndpoint func, there's check of this parameter:
```
        for _, epInfo := range proxier.endpointsMap[svcPortName] {
                if onlyNodeLocalEndpoints && !epInfo.GetIsLocal() {
                        continue
                }
                newEndpoints.Insert(epInfo.String())
        }
```

And I manually tested accessing NodePort with externalTrafficPolicy=Cluster and externalTrafficPolicy=Local.  The random timeout disappeared after this fix. 

When externalTrafficPolicy=Local, without this fix, all endpoint addresses are added as RS(checed by ipvsadm -Ln).  Only **local** endpoint addresses  are added as RS.

**Comment by m1093782566** [comment]
Okay, thanks for clarification. I was just missed :)

**Comment by m1093782566** [comment]
And you probably should fix the test as well?

**Comment by m1093782566** [comment]
go fmt please.

**Comment by honkiko** [comment]
updated for gofmt

**Pull request merged**